### PR TITLE
LDAP authentication: support active role service

### PR DESCRIPTION
**Pull request by NielsCharlier**

### CODE REVIEW

File: src/security/ldap/src/main/java/org/geoserver/security/ldap/LDAPRoleService.java [hunk 1]
```diff
@@ -222,7 +222,7 @@ public void executeWithContext(DirContext ctx,
                         String dn = LDAPUtils
                                 .getLdapTemplateInContext(ctx, template)
                                 .searchForSingleEntry("", userFilter,
-                                        new String[] { username }).getDn()
+                                        new String[] { username }).getNameInNamespace()
                                 .toString();
```
Comment (mbarto) [inline src/security/ldap/src/main/java/org/geoserver/security/ldap/LDAPRoleService.java]: Is this needed?

**Comment by mbarto** [comment]
I think some unit-tests are needed

File: src/security/ldap/src/main/java/org/geoserver/security/ldap/LDAPRoleService.java (hunk 1 above)
Comment (NielsCharlier) [inline src/security/ldap/src/main/java/org/geoserver/security/ldap/LDAPRoleService.java]: As I commented on the commit diff, this change actually has nothing to do with the bug. But in the auth.prov. the parameter {2} applies to the full DN while in the role service it applies to the dn without base. I wanted to make this consistent with each other. So {2} always means the same thing

Comment (NielsCharlier) [inline src/security/ldap/src/main/java/org/geoserver/security/ldap/LDAPRoleService.java]: I removed this change from the patch and kept it one issue focused. This change was causing problems in the pattern matching of user DNs so it was going to take me too far. But I would like to make a note for the future that ideally there should be consistency between the two ldap services.

**Comment by NielsCharlier** [comment]
Unit test of functionality added

**Comment by jodygarnett** [comment]
This is a bug fix, do we need a backport to 2.6.x as well?

**Comment by mbarto** [comment]
Looks good to me, now

**Comment by mbarto** [comment]
For me it's good to merge. Do you have permissions do it by yourself, Niels, or do you need someone to do it for you? (In this case I can do it)

**Comment by jodygarnett** [comment]
In general if you have time to do the review please merge :)

--
Jody Garnett

**Pull request merged**